src/handler.js

```diff
@@ -39,18 +39,42 @@ const getReplyToFromRequestHeaders = (request) => {
 }
 
 const createFSPIOPErrorFromErrorResponse = (request, response) => {
-  const fspiopError = ((httpStatusCode) => {
-    switch (httpStatusCode) {
-      case 400:
-        return Errors.CLIENT_ERROR
-
-      case 404:
-        return Errors.UNKNOWN_URI
+  // The content of this function is a naive attempt at encoding a complex set of conditions
+  // declaratively, without good knowledge of future requirements. Put another way: if this becomes
+  // unwieldly, there's no reason it shouldn't be changed.
+  const getBadRequestMessage = inputMessage => {
+    const badRequestMessages = [
+      {
+        // Look for invalid headers and return a MALFORMED_SYNTAX error
+        regex: /^Invalid.*header$/,
+        code: Errors.MALFORMED_SYNTAX
+      },
+      {
+        // When the user supplies an unacceptable version in the Accept header
+        regex: /^Unacceptable version requested$/,
```

Comment:
Why not rather throw an FSPIOPError on your version validation plugin instead of modifying this method/function?